int8[] num 
int8[] rgb
